src/causalprog/distribution/family.py

```diff
@@ -67,13 +69,33 @@ def __init__(
         self._family = backend_family
         self._family_translator = backend_translator
 
-    def construct(self, **parameters: ArrayLike) -> Distribution:
+    def construct(self, **parameters: npt.ArrayLike) -> Distribution:
         """
         Create a distribution from an explicit set of parameters.
 
         Args:
-            **parameters (ArrayLike): Parameters that define a member of this family,
+            **parameters: Parameters that define a member of this family,
                 passed as sequential arguments.
 
         """
         return self._member(**parameters)
+
+    def sample(
+        self,
+        samples: int,
+        rng_key: jax.Array,
+        **kwargs: npt.ArrayLike,
+    ) -> npt.NDArray[float]:
+        """Sample values from the distribution."""
+        raise NotImplementedError
+
+        output = np.zeros(samples)
+        new_key = jax.random.split(rng_key, samples)
+        for sample in range(samples):
+            parameters = {
+                i: j[sample] if hasattr(j, "__len__") else j for i, j in kwargs.items()
+            }
```

Comment:
```suggestion
            parameters = {
                param_name: param_sample[sample] if hasattr(param_sample, "__len__") else param_sample for param_name, param_sample in kwargs.items()
            }
```

Just so it's a bit less obtuse what's going on here now (since we are mixing scalar and array-like samples now).

In the long-term we should probably decide on a more elegant way to handle the scalar/array issue than checking for `__len__`. Though it might be better to test for the `__getitem__` attribute (which allows use of the access `[ ]` syntax on objects) rather than `__len__` as some containers have a length but don't support accessing.